gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -330,10 +388,38 @@ private MapGeoJson toMapGeoJson(GeoJsonFeature feature) {
   }
 
   public LiveData<CameraPosition> getCameraPosition() {
+    checkPointNearVertex(cameraPosition.getValue());
     Timber.d("Current position is %s", cameraPosition.getValue().toString());
     return cameraPosition;
   }
 
+  private void checkPointNearVertex(CameraPosition position) {
+    if (vertices.isEmpty() || vertices.size() < 3) {
+      return;
+    }
+    //  Experimentation with isPointNearFirstVertex() if does not work
+    //  replace vertices.contains(point)
```

Comment:
Not sure I understand this comment. Could you please clarify or remove?

Reply:
removed.